bcgov/cas-registration
# Chore/1862 refine multi-zone links and routing

Addresses:
[1862](https://github.com/bcgov/cas-registration/issues/1862)

## 🚀 Impact:

- Updated dashboard links `bc_obps/common/fixtures/dashboard` 
- Refactored breadcrumb links to accept defaultLinks and app zone.
- Updated `bciers/libs/shared/components/src/lib/layout/RootLayout.tsx` to implement refactored breadcrumbs
- Updated mulit-zone apps'  `layout.tsx` to implement refactored breadcrumbs
- Updated dashboard routing middleware `bciers/apps/dashboard/middlewares/withAuthorizationDashboard.tsx`
- Updated registration routing middleware `bciers/apps/registration/middlewares/withAuthorizationRegistration.tsx`
- Updated reporting routing middleware `bciers/apps/reporting/middlewares/withAuthorizationReporting.tsx`
- Updated reporting dashboard page `bciers/apps/reporting/src/app/page.tsx`
- Update profile link `bciers/libs/shared/components/src/lib/navigation/Profile.tsx`



## 🧭 Architectural Changes:

- Localized some components from `registration1` to `registration`
- Created shared libs for components and utils used by `registration` components
- Updated  `registration` imports from `@app` to `@bciers`
- Updated  shared imports from `@app` to `@bciers`

## 🧹 Clean-up:

- Deleted `dashboard` folder originally used for dynamic breadcrumbs
- Deleted `bciers/apps/dashboard/app/components/auth`
- Deleted `bciers/apps/dashboard/app/data`
- Deleted `bciers/apps/registration/app/page.tsx`



##  🔬 Local Testing:

From terminal command, start the api server:
 ```
cd bc_obps
make reset_db
make run
```  
From terminal command, start the `BCiers dashboard` Nx Next.js app
 ```
cd bciers && yarn dev-all
```  
**Testing industry_user_admin**
1. Navigate to `http://localhost:3000`
 **Expected result**: redirection to  `http://localhost:3000/onboarding`
![image](https://github.com/bcgov/cas-registration/assets/120038448/306585b2-d643-4227-a3ce-4310e0c169fb)
2. Navigate to `http://localhost:3000/dashboard`
 **Expected result**: redirection to  `http://localhost:3000/onboarding`
3. Navigate to `http://localhost:3000/registration`
 **Expected result**: redirection to  `http://localhost:3000/onboarding`
4. Navigate to `http://localhost:3000/reporting`
 **Expected result**: redirection to  `http://localhost:3000/onboarding`
5. Click button "Log in with Business BCeID" 
6. Sign in to Keycloak using `bc-cas-dev`
**Expected result**: redirection to `http://localhost:3000/dashboard` page for role `industry_user_admin`
![image](https://github.com/bcgov/cas-registration/assets/120038448/ca4dfeae-f1d4-4bb5-8c9c-3d73839b45e0)
Breadcrumbs are correct
7. Click link `Registration\My Operator`
 **Expected result**: redirection to `http://localhost:3000/registration/select-operator`
![image](https://github.com/bcgov/cas-registration/assets/120038448/0443b55e-bba2-414b-ac05-001d0e80a9c9)
Breadcrumbs are correct
8, Click `Dashboard`  breadcrumb
**Expected result**: redirection to `http://localhost:3000/dashboard` 
Breadcrumbs are correct
9. Click link `Registration\Operations`
**Expected result**: redirection to `http://localhost:3000/registration/operations`
![image](https://github.com/bcgov/cas-registration/assets/120038448/3b5d9bcc-1ad1-4318-8e98-a611b2f569f2)
Breadcrumbs are correct
10. Click `Add Operation` button
 **Expected result**: redirection to `http://localhost:3000/registration/operations/create/1`
![image](https://github.com/bcgov/cas-registration/assets/120038448/6d386500-8a6b-49f8-b71b-4019c1eca5d3)
Breadcrumbs are correct
11. Click link `Operations` breadcrumb
**Expected result**: redirection to `http://localhost:3000/registration/operations`
![image](https://github.com/bcgov/cas-registration/assets/120038448/3b5d9bcc-1ad1-4318-8e98-a611b2f569f2)
Breadcrumbs are correct
12. Click button "Log Out" 
 **Expected result**: redirection to  keycloak logged out



**Testing industry_user**
1. Navigate to `http://localhost:3000`
 **Expected result**: redirection to  `http://localhost:3000/onboarding`
2. Sign in to Keycloak using `bc-cas-dev-secondary`
**Expected result**: redirection to `http://localhost:3000/dashboard` page for role `industry_user`
![image](https://github.com/bcgov/cas-registration/assets/120038448/0b6e6c75-814d-4d55-83f5-2a0397ba5354)
Breadcrumbs are correct
3. Click link `Registration\My Operator`
 **Expected result**: redirection to `http://localhost:3000/registration/select-operator`
![image](https://github.com/bcgov/cas-registration/assets/120038448/b76f52e8-81ce-4523-a712-2c6ac11e0b40)
Breadcrumbs are correct
4. Click `Add Operation` button
 **Expected result**: redirection to `http://localhost:3000/registration/select-operator/user-operator/create/1`
![image](https://github.com/bcgov/cas-registration/assets/120038448/af7f4206-7fdd-4bae-a055-2f685435d59c)
Breadcrumbs are correct
5. Click `Select Operator` breadcrumb
 **Expected result**: redirection to `http://localhost:3000/registration/select-operator`
![image](https://github.com/bcgov/cas-registration/assets/120038448/b76f52e8-81ce-4523-a712-2c6ac11e0b40)
Breadcrumbs are correct
6. Use the search to select an operator from the list
![image](https://github.com/bcgov/cas-registration/assets/120038448/32ac1cf7-ed11-42b8-a7bd-ecc3045efe35)
7. Click `Select Operator`
 **Expected result**: redirection to `http://localhost:3000/registration/select-operator/confirm/`
![image](https://github.com/bcgov/cas-registration/assets/120038448/94641f73-950b-4891-a169-68e924f9bc0c)
8. Click `Return` 
9. **Expected result**: redirection to `http://localhost:3000/registration/select-operator`
![image](https://github.com/bcgov/cas-registration/assets/120038448/b76f52e8-81ce-4523-a712-2c6ac11e0b40)
10. Use the search to select an operator from the list
11. Click `Select Operator`
 **Expected result**: redirection to `http://localhost:3000/registration/select-operator/confirm/`
![image](https://github.com/bcgov/cas-registration/assets/120038448/94641f73-950b-4891-a169-68e924f9bc0c)
12. Click "Yes, this is my Operator"
**Expected result**: 
![image](https://github.com/bcgov/cas-registration/assets/120038448/be8aa38c-d676-4241-b224-9afbe1eeffda)
13. Click "Go Back" button
 **Expected result**:
![image](https://github.com/bcgov/cas-registration/assets/120038448/94641f73-950b-4891-a169-68e924f9bc0c)
14. Click "Yes, this is my Operator"
**Expected result**: 
![image](https://github.com/bcgov/cas-registration/assets/120038448/be8aa38c-d676-4241-b224-9afbe1eeffda)
15 . Click `Request Administrator Access`
**Expected result**: 
![image](https://github.com/bcgov/cas-registration/assets/120038448/946bda7c-d09b-4d85-ab45-d1be2b7e1924)
16. Click button "Log Out" 
 **Expected result**: redirection to  keycloak logged out
Note: `Operations` flow not tested as the middleware returns to dashboard ` if (
              operator.status !== "Pending" &&
              operator.status !== "Approved"
            )`


**Testing new user**
1. Navigate to `http://localhost:3000`
2. Sign in to Keycloak using `bc-cas-dev-three`
**Expected result**: redirection to `http://localhost:3000/registration/profile` page for no role
![image](https://github.com/bcgov/cas-registration/assets/120038448/33b293a8-9989-4ddc-a39b-8966f355e1cf)
